test/mocks.ts

```diff
@@ -31,7 +34,10 @@ export function createMockAppWindow(): jest.Mocked<AppWindow> {
         bringToFront: jest.fn<Promise<void>, []>(),
         focus: jest.fn<Promise<void>, []>(),
         isReadyToReceiveIntent: jest.fn<Promise<boolean>, [IntentType]>(),
-        removeAllListeners: jest.fn<void, []>()
+        isReadyToReceiveContext: jest.fn<Promise<boolean>, []>(),
+        removeAllListeners: jest.fn<void, []>(),
+        // Apply any custom overrides
+        ...options
```

Comment:
Hmm. Could we do this everywhere or nowhere?

We've got a few instances of eg. `{...createMockChannel(), id: 'test-3', type: 'system'}` which takes the inverse approach.

Reply:
Sure. Makes sense to keep it consistent.